Classes/Domain/Index/TcaIndexer/TcaTableService76.php

```diff
@@ -0,0 +1,379 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use Codappix\SearchCore\Configuration\ConfigurationContainerInterface;
+use Codappix\SearchCore\Domain\Index\IndexingException;
+use Codappix\SearchCore\Domain\Index\TcaIndexer\InvalidArgumentException;
+use TYPO3\CMS\Backend\Utility\BackendUtility;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\RootlineUtility;
+use TYPO3\CMS\Extbase\Object\ObjectManagerInterface;
+
+/**
+ * Encapsulate logik related to TCA configuration.
+ */
+class TcaTableService76 implements TcaTableServiceInterface
+{
+    /**
+     * TCA for current table.
+     * !REFERENCE! To save memory.
+     * @var array
+     */
+    protected $tca;
+
+    /**
+     * @var string
+     */
+    protected $tableName;
+
+    /**
+     * @var ConfigurationContainerInterface
+     */
+    protected $configuration;
+
+    /**
+     * @var RelationResolver
+     */
+    protected $relationResolver;
+
+    /**
+     * @var \TYPO3\CMS\Core\Log\Logger
+     */
+    protected $logger;
+
+    /**
+     * @var ObjectManagerInterface
+     */
+    protected $objectManager;
+
+    /**
+     * Inject log manager to get concrete logger from it.
+     *
+     * @param \TYPO3\CMS\Core\Log\LogManager $logManager
+     */
+    public function injectLogger(\TYPO3\CMS\Core\Log\LogManager $logManager)
+    {
+        $this->logger = $logManager->getLogger(__CLASS__);
+    }
+
+    /**
+     * @param ObjectManagerInterface $objectManager
+     */
+    public function injectObjectManager(ObjectManagerInterface $objectManager)
+    {
+        $this->objectManager = $objectManager;
+    }
+
+    /**
+     * @param string $tableName
+     * @param ConfigurationContainerInterface $configuration
+     */
+    public function __construct(
+        $tableName,
+        RelationResolver $relationResolver,
+        ConfigurationContainerInterface $configuration
+    ) {
+        if (!isset($GLOBALS['TCA'][$tableName])) {
+            throw new IndexingException(
+                'Table "' . $tableName . '" is not configured in TCA.',
+                IndexingException::CODE_UNKOWN_TCA_TABLE
+            );
+        }
+
+        $this->tableName = $tableName;
+        $this->tca = &$GLOBALS['TCA'][$this->tableName];
+        $this->configuration = $configuration;
+        $this->relationResolver = $relationResolver;
+    }
+
+    public function getTableName() : string
+    {
+        return $this->tableName;
+    }
+
+    public function getTableClause() : string
+    {
+        if ($this->tableName === 'pages') {
```

Comment:
Why is this part different from the 8.7 part?

https://github.com/Codappix/search_core/blob/feature/support-typo3-7-to-8/Classes/Domain/Index/TcaIndexer/TcaTableService.php#L118

Reply:
Because in 8 we have doctrine and use `getJoins` to fetch the join information